pkg/logger/formatter.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	colorRed    = 31
+	colorYellow = 33
+	colorBlue   = 36
+	colorGray   = 37
+)
+
+const (
+	defaultTimestampFormat = "2006-01-02 15:03:04"
+)
+
+func getColorByLevel(level logrus.Level) int {
+	switch level {
+	case logrus.DebugLevel, logrus.TraceLevel:
+		return colorGray
+	case logrus.WarnLevel:
+		return colorYellow
+	case logrus.ErrorLevel, logrus.FatalLevel, logrus.PanicLevel:
+		return colorRed
+	default:
+		return colorBlue
+	}
+}
+
+type Formatter struct {
```

Comment:
Why need provide a new Formatter?

Why don't use logrus.TextFormatter?

Reply:
the main purpose of using ` new Formatter` is format the output message with file and line number: `2022-07-04 18:06:36 [INFO] [local.go:163] Start to scale this cluster", and it only used in console output, having no effect on others.

Reply:
ok

Reply:
`INFO[0000] No upgrade required, image version and cluster version are both v1.19.8.`